### DISCUSSION TITLE
Vertical text alignment issue with word-wrap

### DISCUSSION THREAD

**Discussion by a-erlandsen-z** [slint-ui/slint#3479] [Q&A]
So I want to vertically align text within a box.
I also want to have `word-wrap` and `TextOverflow.elide`.
The issue I have is this: When the text is short enough to fit within the box, text is vertically aligned when vertical alignment is set to `TextVerticalAlignment.center`. But when the text is too long (and "..." is added at the end), text is only vertically aligned when vertical-alignment is set to `TextVerticalAlignment.top`.

**Seems like a bug to me?**

Here is a sample of what I'm doing:

    StatusBox := Rectangle {
        clip: true;
      
        StatusText := Text {
            text: "Call is being transferred Call is being transferred Call is being transferred";
            
            font-size: 14px;
            font-weight: 300;
            
            height: 100%;
            width: 100%;
            overflow: TextOverflow.elide;
            wrap: word-wrap;
            vertical-alignment: TextVerticalAlignment.center;
        }
    }

This will align poorly (first line is moved up and be clipped at the top). To get this properly vertically aligned, either the text needs to be shorter, or vertical alignment needs to be set to `top`. But if set to `top` it will not be properly aligned when text is shorter. And the whole point here is that the text is dynamic (different languages), so the length of the text string will vary.

**Comment by ogoffart** [comment]
Thanks for reporting this issue, this is clearly a bug.
Created https://github.com/slint-ui/slint/issues/3481  to track it

**Comment by ogoffart** [comment]
We just released Slint 1.2.1 that should produce hopefully what you want in this case. (Especially https://github.com/slint-ui/slint/pull/3488)

**Reply by a-erlandsen-z** [reply]
Perfect. Thanks for the quick turnaround.
